commit 64bf6d9705aa8724a17a137e76dc6f5b58309026
Author: Jiri Olsa <jolsa@redhat.com>
Date:   Sat Feb 27 21:21:12 2016 +0100

    perf tools: Fix python extension build
    
    commit 67d5268908283c187e0a460048a423256c2fb288 upstream.
    
    The util/python-ext-sources file contains source files required to build
    the python extension relative to $(srctree)/tools/perf,
    
    Such a file path $(FILE).c is handed over to the python extension build
    system, which builds the final object in the
    $(PYTHON_EXTBUILD)/tmp/$(FILE).o path.
    
    After the build is done all files from $(PYTHON_EXTBUILD)lib/ are
    carried as the result binaries.
    
    Above system fails when we add source file relative to ../lib, which we
    do for:
    
      ../lib/bitmap.c
      ../lib/find_bit.c
      ../lib/hweight.c
      ../lib/rbtree.c
    
    All above objects will be built like:
    
      $(PYTHON_EXTBUILD)/tmp/../lib/bitmap.c
      $(PYTHON_EXTBUILD)/tmp/../lib/find_bit.c
      $(PYTHON_EXTBUILD)/tmp/../lib/hweight.c
      $(PYTHON_EXTBUILD)/tmp/../lib/rbtree.c
    
    which accidentally happens to be final library path:
    
      $(PYTHON_EXTBUILD)/lib/
    
    Changing setup.py to pass full paths of source files to Extension build
    class and thus keep all built objects under $(PYTHON_EXTBUILD)tmp
    directory.
    
    Reported-by: Jeff Bastian <jbastian@redhat.com>
    Signed-off-by: Jiri Olsa <jolsa@kernel.org>
    Tested-by: Josh Boyer <jwboyer@fedoraproject.org>
    Cc: David Ahern <dsahern@gmail.com>
    Cc: Namhyung Kim <namhyung@kernel.org>
    Cc: Peter Zijlstra <a.p.zijlstra@chello.nl>
    Link: http://lkml.kernel.org/r/20160227201350.GB28494@krava.redhat.com
    Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/tools/perf/util/setup.py b/tools/perf/util/setup.py
index 1833103..c868098 100644
--- a/tools/perf/util/setup.py
+++ b/tools/perf/util/setup.py
@@ -22,6 +22,7 @@ cflags = getenv('CFLAGS', '').split()
 # switch off several checks (need to be at the end of cflags list)
 cflags += ['-fno-strict-aliasing', '-Wno-write-strings', '-Wno-unused-parameter' ]
 
+src_perf  = getenv('srctree') + '/tools/perf'
 build_lib = getenv('PYTHON_EXTBUILD_LIB')
 build_tmp = getenv('PYTHON_EXTBUILD_TMP')
 libtraceevent = getenv('LIBTRACEEVENT')
@@ -30,6 +31,9 @@ libapikfs = getenv('LIBAPI')
 ext_sources = [f.strip() for f in file('util/python-ext-sources')
 				if len(f.strip()) > 0 and f[0] != '#']
 
+# use full paths with source files
+ext_sources = map(lambda x: '%s/%s' % (src_perf, x) , ext_sources)
+
 perf = Extension('perf',
 		  sources = ext_sources,
 		  include_dirs = ['util/include'],
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1259_64bf6d9705aa8724a17a137e76dc6f5b58309026.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1259_64bf6d9705aa8724a17a137e76dc6f5b58309026.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
